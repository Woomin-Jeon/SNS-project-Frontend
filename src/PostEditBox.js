/* eslint-disable no-alert */
/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
import React, { useState } from 'react';

function PostEditBox({
  specificPost,
  postState,
  setPostState,
  currentUserState,
}) {
  const [temptState, setTemptState] = useState('');
  const { post } = postState;
  const { id } = currentUserState;

  const setEditBox = (temptState) => {
    setTemptState(temptState);
  };

  const handleEditPostContents = () => {
    if (temptState.trim()) {
      if (specificPost.id === id) {
        setPostState({
          ...postState,
          post: post.map((p) =>
            (p.uniqueKey !== specificPost.uniqueKey ? p : { ...p, contents: temptState })),
        });
        setTemptState('');
        alert('게시글이 수정되었습니다. "수정" 버튼을 눌러서 창을 닫아주세요');
      } else {
        alert('게시글은 해당 작성자만 수정 할 수 있습니다');
      }
    } else {
      alert('수정사항을 입력해주세요');
    }
  };

  return (
    <div>
      <input
        className="post-edit-box"
        type="text"
        value={temptState}
        onChange={(e) => setEditBox(e.target.value)}
      />
      <button
        className="post-edit-complete"
        type="button"
        onClick={handleEditPostContents}
      >
      완료
      </button>
    </div>
  );
}

export default PostEditBox;
