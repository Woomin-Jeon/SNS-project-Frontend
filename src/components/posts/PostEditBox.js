import React, { useState } from 'react';
import func from '../../function';
import Swal from "sweetalert2";

function PostEditBox({
  specificPost,
  postState,
  setPostState,
  currentUserState,
}) {
  const [temptState, setTemptState] = useState('');
  const { id } = currentUserState;

  const setEditBox = (temptState) => {
    setTemptState(temptState);
  };

  const clickedEditSuccess = async () => {
    const { uniqueKey } = specificPost;

    if (!temptState.trim()) {
      await Swal.fire('', '수정사항을 입력해주세요', 'error');
      return;
    }

    if (specificPost.id !== id) {
      await Swal.fire('', '게시글은 해당 작성자만 수정 할 수 있습니다', 'error');
      return;
    }

    const { timeLinePosts } = await func.editPost(uniqueKey, temptState);
    setPostState({ ...postState, post: [...timeLinePosts.reverse()] });
    await Swal.fire('', '게시글이 수정되었습니다', 'success');
  };

  return (
    <div>
      <textarea
        className="showpost-edit-box"
        type="text"
        value={temptState}
        onChange={(e) => setEditBox(e.target.value)}
      />
      <button
        className="showpost-edit-complete"
        type="button"
        onClick={clickedEditSuccess}
      >
      완료
      </button>
    </div>
  );
}

export default PostEditBox;
