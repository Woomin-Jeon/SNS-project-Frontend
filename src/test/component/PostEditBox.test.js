import React from 'react';
import {shallow} from "enzyme";
import PostEditBox from "../../components/posts/PostEditBox";

describe('PostEditBox', () => {
  let specificPost;
  let postState;
  let setPostState;
  let currentUserState;

  beforeEach(() => {
    currentUserState = { id: 'Random ID' }
  });

  it('renders well', () => {
    const component = shallow(
      <PostEditBox
        specificPost={specificPost}
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('button').text()).toBe('완료');
    expect(component.find('textarea')).toBeDefined();
  });
});
