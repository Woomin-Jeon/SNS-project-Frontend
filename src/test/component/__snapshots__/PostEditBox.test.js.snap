// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostEditBox renders well 1`] = `ShallowWrapper {}`;
